src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -76,7 +76,10 @@ internal sealed class HttpConnectionPool : IDisposable
         private byte[]? _http2AltSvcOriginUri;
         internal readonly byte[]? _http2EncodedAuthorityHostHeader;
 
-        private readonly bool _http3Enabled;
+        [SupportedOSPlatformGuard("linux")]
+        [SupportedOSPlatformGuard("macOS")]
+        [SupportedOSPlatformGuard("Windows")]
+        private readonly bool _http3Enabled = (OperatingSystem.IsLinux() && !OperatingSystem.IsAndroid()) || OperatingSystem.IsWindows() || OperatingSystem.IsMacOS();
```

Comment:
I still think the condition logic should be in IsHttp3Supported method so it does not need to be duplicated in the file

Reply:
This is not right IMHO, please do not change the existing logic of `_http3Enabled` and add addition `IsHttp3Supported` as Marek suggests.